src/Cookie.php

```diff
@@ -0,0 +1,494 @@
+    {
+        return isset($this->expires) and $this->expires < new \DateTime();
+    }
+
+    /**
+     * Returns the domain.
+     *
+     * @return string|null
+     */
+    public function getDomain()
+    {
+        return $this->domain;
+    }
+
+    /**
+     * Checks if there is a domain.
+     *
+     * @return bool
+     */
+    public function hasDomain()
+    {
+        return isset($this->domain);
+    }
+
+    /**
+     * Sets the domain.
+     *
+     * @param string|null $domain
+     *
+     * @return self
+     */
+    public function withDomain($domain)
+    {
+        $new = clone $this;
+        $new->domain = $this->normalizeDomain($domain);
+
+        return $new;
+    }
+
+    /**
+     * Matches a domain.
+     *
+     * @param string $domain
+     *
+     * @return bool
+     *
+     * @see http://tools.ietf.org/html/rfc6265#section-5.1.3
+     */
+    public function matchDomain($domain)
+    {
+        // Domain is not set or exact match
+        if (!$this->hasDomain() || strcasecmp($domain, $this->domain) === 0) {
+            return true;
+        }
+
+        // Domain is not an IP address
+        if (filter_var($domain, FILTER_VALIDATE_IP)) {
+            return false;
+        }
+
+        return (bool) preg_match('/\b'.preg_quote($this->domain).'$/i', $domain);
+    }
+
+    /**
+     * Returns the path.
+     *
+     * @return string
+     */
+    public function getPath()
+    {
+        return $this->path;
+    }
+
+    /**
+     * Sets the path.
+     *
+     * @param string|null $path
+     *
+     * @return self
+     */
+    public function withPath($path)
+    {
+        $new = clone $this;
+        $new->path = $this->normalizePath($path);
+
+        return $new;
+    }
+
+    /**
+     * It matches a path.
+     *
+     * @param string $path
+     *
+     * @return bool
+     *
+     * @see http://tools.ietf.org/html/rfc6265#section-5.1.4
+     */
+    public function matchPath($path)
+    {
+        return $this->path === $path || (strpos($path, $this->path.'/') === 0);
+    }
+
+    /**
+     * Checks if HTTPS is required.
+     *
+     * @return bool
+     */
+    public function isSecure()
+    {
+        return $this->secure;
+    }
+
+    /**
+     * Sets the secure.
+     *
+     * @param bool $secure
+     *
+     * @return self
+     */
+    public function withSecure($secure)
+    {
+        $new = clone $this;
+        $new->secure = (bool) $secure;
+
+        return $new;
+    }
+
+    /**
+     * Checks if it is HTTP-only.
+     *
+     * @return bool
+     */
+    public function isHttpOnly()
+    {
+        return $this->httpOnly;
+    }
+
+    /**
+     * Sets the HTTP Only.
+     *
+     * @param bool $httpOnly
+     *
+     * @return self
+     */
+    public function withHttpOnly($httpOnly)
+    {
+        $new = clone $this;
+        $new->httpOnly = (bool) $httpOnly;
+
+        return $new;
+    }
+
+    /**
+     * Checks if it matches with another cookie.
```

Comment:
Checks if this cookie represents the same cookie as $cookie.

This does not compare the values, only name, domain and path.